docs/user-guide/private-repositories.md

```diff
@@ -43,13 +43,90 @@ argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-ke
 
 ## Self-signed & Untrusted TLS Certificates
 
+> v1.2 or higher
+
+If you are connecting a repository on a HTTPS server using a self-signed certificate, or a certificate signed by a custom Certificate Authority (CA) which are not known to ArgoCD, the repository will not be added due to security reasons. This is indicated by an error message such as `x509: certificate signed by unknown authority`.
+
+1. You can let ArgoCD connect the repository in an insecure way, without verifying the server's certificate at all. This can be accomplished by using the `--insecure-repository` flag when adding the repository with the `argocd` CLI utility. However, this should be done only for non-production setups, as it imposes a serious security issue through possible man-in-the-middle attacks.
```

Comment:
this is incorrect?

Reply:
I'm unsure about what you mean. What exactly is incorrect?

Reply:
I think @alexec means it supposed to be `insecure-ignore-host-key` instead of `--insecure`